pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+                        for f in o.faces:
+                            if f.id == tool.id:
+                                obj_name = o.name
+                                obj = f
+                    tool_type = "FaceTool"
+                elif isinstance(tool, EdgePrimitive):
+                    for o in self.object_list:
+                        for e in o.edges:
+                            if e.id == tool.id:
+                                obj_name = o.name
+                                obj = e
+                    tool_type = "EdgeTool"
+                elif isinstance(tool, Polyline) or tool.object_type != "Line":
+                    for o in self.object_list:
+                        if o.name == tool.name:
+                            obj_name = tool.name
+                            obj = o.edges[0]
+                    tool_type = "EdgeTool"
+                else:
+                    self.logger.error("Face tool part has to be provided as a string (name) or an int (face id).")
+                    return False
+                planes = "Dummy"
+                tool_type = tool_type
+                tool_entity_id = obj.id
+                selections = [
+                    "NAME:Selections",
+                    "Selections:=",
+                    objects,
+                    "NewPartsModelFlag:=",
+                    "Model",
+                    "ToolPart:=",
+                    obj_name,
+                ]
+        else:
+            if not plane and tool or not plane:
+                self.logger.info("For 2D design types only planes can be defined.")
+                return False
+            elif plane:
+                tool_type = "PlaneTool"
+                tool_entity_id = -1
+                planes = GeometryOperators.cs_plane_to_plane_str(plane)
+                selections = ["NAME:Selections", "Selections:=", objects, "NewPartsModelFlag:=", "Model"]
+        self.oeditor.Split(
+            selections,
+            [
+                "NAME:SplitToParameters",
+                "SplitPlane:=",
+                planes,
+                "WhichSide:=",
+                sides,
+                "ToolType:=",
+                tool_type,
+                "ToolEntityID:=",
+                tool_entity_id,
+                "SplitCrossingObjectsOnly:=",
+                split_crossing_objs,
+                "DeleteInvalidObjects:=",
+                delete_invalid_objs,
+            ],
+        )
+        self.refresh_all_ids()
+        return [objects] + [i for i in self.object_names if i not in all_objs]
+
+    @pyaedt_function_handler()  # TODO: Deprecate this and add duplicate as an option in the mirror method.
+    def duplicate_and_mirror(
+        self,
+        objid,
+        position,
+        vector,
+        is_3d_comp=False,
+        duplicate_assignment=True,
+    ):
+        """Duplicate and mirror a selection.
+
+        Parameters
+        ----------
+        objid : str, int, or  Object3d
```

Comment:
```suggestion
        objid : str, int, or Object3d
```
Does ``Object3D`` need to included somehow in the description?